lib/still/compiler/incremental/node.ex

```diff
@@ -102,109 +71,58 @@ defmodule Still.Compiler.Incremental.Node do
   end
 
   @impl true
-  def handle_call(:compile, _from, %{cached_source_file: source_file} = state)
+  def handle_call({_, use_cache: true}, _from, %{cached_source_file: source_file} = state)
       when not is_nil(source_file) do
     {:reply, source_file, state}
   end
 
-  def handle_call(:compile, _from, state) do
-    case Compile.run(state) do
-      {:ok, source_file} ->
-        ErrorCache.set({:ok, source_file})
-        {:reply, source_file, %{state | cached_source_file: source_file}}
-
-      other ->
-        {:reply, other, state}
-    end
-  catch
-    _, %PreprocessorError{} = error ->
-      handle_compile_error(error)
+  def handle_call({:compile, _}, from, state) do
+    froms = all_waiting_compile([from])
 
-      {:reply, :ok, state}
+    try do
+      source_file = __MODULE__.Compile.run(state.file)
 
-    kind, payload ->
-      error = %PreprocessorError{
-        payload: payload,
-        kind: kind,
-        stacktrace: __STACKTRACE__,
-        source_file: %Still.SourceFile{input_file: state.file, run_type: :compile}
-      }
+      Enum.each(froms, &GenServer.reply(&1, source_file))
 
-      handle_compile_error(error)
+      {:noreply, %{state | cached_source_file: source_file}}
+    catch
+      _, %PreprocessorError{} ->
+        Enum.each(froms, &GenServer.reply(&1, :ok))
 
-      {:reply, :ok, state}
+        {:noreply, state}
+    end
   end
 
   @impl true
-  def handle_call({:render, data, nil}, _from, state) do
-    do_render(data, state)
+  def handle_call({:render, data}, _from, state) do
+    source_file = __MODULE__.Render.run(state.file, data)
+
+    {:reply, source_file, state}
+  catch
+    _, %PreprocessorError{} = error ->
+      {:reply, error, state}
   end
 
   @impl true
-  def handle_call({:render, data, subscriber}, _from, state) do
-    subscribers =
-      [subscriber | state.subscribers]
-      |> Enum.uniq()
-      |> Enum.reject(&is_nil/1)
+  def handle_call({:compile_metadata, _opts}, _from, state) do
+    source_file = __MODULE__.Compile.run(state.file, :compile_metadata)
 
-    do_render(data, %{state | subscribers: subscribers})
+    {:reply, source_file, %{state | cached_source_file: source_file}}
+  catch
+    _, %PreprocessorError{} ->
```

Comment:
Shouldn't this be logged as well?

Reply:
Nope, that's already handled in the error flow. It happens before it reaches this part